### ISSUE TITLE
MobX 5.9.0 support

### ISSUE THREAD

**Issue by romandubrov** [mobxjs/mobx-state-tree#1143]
**_Question_**

* [x] I've checked documentation and searched for existing issues
* [ ] I tried the [spectrum channel](https://spectrum.chat/?t=dad48299-3dfc-4e10-b6da-9af1e39498a3)

unlike to previous versions, mobx@5.9.0 has "shapshot" property is not configurable anymore, so MST crashes when trying to replace it when deleting a node.

```
/* compiled mobx.module.js */
function generateComputedPropConfig$$1(propName) {
  return computedPropertyConfigs[propName] || (computedPropertyConfigs[propName] = {
    configurable: false, // <========================== look here
    enumerable: false,
    get: function get() {
      return getAdministrationForComputedPropOwner(this).read(propName);
    },
    set: function set(v) {
      getAdministrationForComputedPropOwner(this).write(propName, v);
    }
  });
}
/* */
```

any compatibility plans? currently had to freese mobx version to 5.8.0 in my package.json

**Comment by xaviergonz** [comment]
Thanks for reporting this! a PR is already in place, so it should be fixed soon

**Comment by xaviergonz** [comment]
Should be fixed in 3.10.1

**Comment by frankLife** [comment]
is there any fix or workaround about 2.x version `mobx-state-tree` ?
